ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiUserManagementService.scala

```diff
@@ -259,26 +261,32 @@ object ApiUserManagementService {
         decodedBytes <- Try[Array[Byte]](Base64.getUrlDecoder.decode(bytes))
           .map(Right(_))
           .recover { case _: IllegalArgumentException =>
-            Left(
-              LedgerApiErrors.RequestValidation.InvalidArgument
-                .Reject("Invalid page token")
-                .asGrpcError
-            )
+            Left(invalidToken)
+          }
+          .get
+        tokenPayload <- Try[ListUsersPageTokenPayload] {
+          ListUsersPageTokenPayload.parseFrom(decodedBytes)
+        }.map(Right(_))
+          .recover { case _: InvalidProtocolBufferException =>
+            Left(invalidToken)
           }
           .get
-        decodedStr = new String(decodedBytes, StandardCharsets.UTF_8)
         userId <- Ref.UserId
-          .fromString(decodedStr)
+          .fromString(tokenPayload.userId)
           .map(Some(_))
           .left
-          .map(_ =>
-            LedgerApiErrors.RequestValidation.InvalidArgument
-              .Reject("Invalid page token")
-              .asGrpcError
-          )
+          .map(_ => invalidToken)
       } yield {
         userId
       }
     }
   }
+
+  private def invalidToken(implicit
```

Comment:
```suggestion
  private def invalidPageToken(implicit
```

avoids confusing wrt the different kinds of tokens in the system.

Reply:
Ok